File: OEM/Domain/Incidents/IncidentDto.cs
using System;

namespace OEMAPI.Domain.Incidents
{
    public class IncidentDto
    {
        public Guid Id { get; set; }
        public string IncidentTypeId { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public string Severity { get; set; }
        public string Description { get; set; }
        public string CreatedBy { get; set; }
    }
}

File: OEM/src/dto/IncidentDTO.ts
export interface IncidentDTO {
  id: string;
  incidentTypeId: string;
  startTime: string;
  endTime?: string;
  severity: string;
  description: string;
  createdBy: string;
}

export interface CreateIncidentDTO {
  incidentTypeId: string;
  startTime: string;
  endTime?: string;
  severity: string;
  description: string;
  createdBy: string;
}

export interface UpdateIncidentDTO {
  incidentTypeId?: string;
  startTime?: string;
  endTime?: string;
  severity?: string;
  description?: string;
  createdBy?: string;
}

// DTO for filtering incidents (US 4.1.13)
export interface IncidentSearchCriteriaDTO {
  vesselName?: string;
  dateStart?: string;
  dateEnd?: string;
  severity?: string;
  status?: 'active' | 'resolved'; // active = no endTime, resolved = has endTime
}